### PR TITLE
TestFramework: Fix coverage

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Part of #8542 

Only the last commit `TestFramework: Fix coverage` needs a review. The rest is here to prove that covered is not 0% anymore

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -35,9 +35,9 @@ variables:
   - name: UnitTestProjectPath
     value: 'analyzers\tests\SonarAnalyzer.Test\'
   - name: UnitTestResultsPath
-    value: '$(Agent.TempDirectory)\TestResults'
+    value: '$(Build.SourcesDirectory)\TestResults'
```
Comment (pavel-mikula-sonarsource) [inline azure-pipelines.yml:38]: Unrelated deconfusion, to use the same directory before and after the artifact is restored

File: azure-pipelines.yml [hunk 2]
```diff
@@ -222,25 +222,25 @@ stages:
       strategy:
         matrix:
           Net48:
-            FrameworkMoniker: 'net48'
-            CoverageArtifactName: 'DotnetCoverageNet48'
-            TestResultsArtifactName: 'DotnetTestResultsNet48'
+            CoverageArtifactName: 'DotNetCoverageNet48'
```
Comment (pavel-mikula-sonarsource) [inline azure-pipelines.yml:225]: Reordered by priority

File: azure-pipelines.yml [hunk 3]
```diff
@@ -251,7 +251,7 @@ stages:
 
       - powershell: |
           cd analyzers
-          & dotnet test $(ProjectFilePath) -f $(FrameworkMoniker) -c $(BuildConfiguration) -l trx --results-directory $(UnitTestResultsPath) /p:AltCover=true,AltCoverForce=true,AltCoverVisibleBranches=true,AltCoverAssemblyFilter='Moq|Humanizer|AltCover|Microsoft.VisualStudio.TestPlatform.*|.*Test',AltCoverPathFilter='SonarAnalyzer\.CFG\\ShimLayer|SonarAnalyzer\.ShimLayer\.CodeGeneration',AltCoverAttributeFilter='ExcludeFromCodeCoverage',AltCoverReport=$(CoveragePath)/coverage.$(FrameworkMoniker).xml
+          & dotnet test $(ProjectFilePath) -f $(FrameworkMoniker) -c $(BuildConfiguration) -l trx --results-directory $(UnitTestResultsPath) /p:AltCover=true,AltCoverForce=true,AltCoverVisibleBranches=true,AltCoverAssemblyFilter="Moq|Humanizer|AltCover|Microsoft.VisualStudio.TestPlatform.*|\.Test^",AltCoverPathFilter="SonarAnalyzer\.CFG\\ShimLayer|SonarAnalyzer\.ShimLayer\.CodeGeneration",AltCoverAttributeFilter="ExcludeFromCodeCoverage",AltCoverReport="$(CoveragePath)/coverage.$(CoverageArtifactName).xml"
```
Comment (pavel-mikula-sonarsource) [inline azure-pipelines.yml:254]: This is the actual fix, `.*Test` was matching the `.TestFramework` part

**Review by mary-georgiou-sonarsource — APPROVED** [review]
LGMT

**Pull request merged**